### DISCUSSION TITLE
RPGLE Languages, outline and content assist

### DISCUSSION THREAD

**Discussion by m-tyler** [codefori/vscode-ibmi#836] [General]
If I perform a change in source, the system triggers off the attempt to rebuild the outline.  When the outline is rebuilding it dumps ALL information and then I have no ability to continue editing with Content Assist, etc.  

For instance, I have snippets that I want to use but as long as the effort to build the outline is active I cannot get the CA to show me any of my snippets.

Is CA tied only to the whole source outline? is there a way to not clear the Outline until the full cross-reference effort on the source is complete?

**Comment by worksofliam** [comment]
> When the outline is rebuilding it dumps ALL information and then I have no ability to continue editing with Content Assist, etc.

Where is this happening?

**Comment by m-tyler** [comment]
I am experiencing this while editing RPG.  What I have found out is that if I Alt-tab from one open tab to another, content assist (CA) does not work right away.  I have to wait several seconds or so for it to appear.   If I edit an RPG member and insert a new line, this causes the outline to rebuild and I am stuck waiting on those server jobs to finish before CA is available.    It appears that VS Code triggered the event to load the outline or something that causes the extension to send jobs to the server to get information. 

If something happens to a SSH server process and it gets stuck `active` the pipeline can clog up, the outline and CA will not appear until this stuck server process is ended.   I have ended stuck server processes and a flood of waiting to process server jobs start and end quickly. 

Again, the CA does not function until whatever is running is done so I can't even get the base CA items like snippets and biffs.

**Reply by worksofliam** [reply]
Weird. The content assist should only be reaching out to IBM i the first time it loads a document, but then only reading from the document until the cache expires (usually around 30 minutes). 

Any video to show this experience you're having?

**Reply by m-tyler** [reply]
No, but I can work on one.  It might take a day or two to get a video that shows it properly.

**Comment by m-tyler** [comment]
I am closing this as I am having difficulty getting a demonstration for you.
